test/utils/setupTruefi2.ts

```diff
@@ -51,7 +53,7 @@ export const setupTruefi2 = async (owner: Wallet, provider: MockProvider, custom
   // ====== DEPLOY ======
   const liquidator = await deployContract(Liquidator2__factory)
   const loanFactory = await deployContract(LoanFactory2__factory)
-  const rater = await deployContract(TrueRatingAgencyV2__factory)
+  const rater: TrueRatingAgencyV2 & TestTrueRatingAgencyV2 = customDeployed?.rater ? customDeployed.rater : await deployContract(TrueRatingAgencyV2__factory)
```

Comment:
Are you sure we want intersection of the types (`&`) and not union (`|`)?

Reply:
No, I'm not sure. I copied what's below and mirrored it.

Reply:
Huh, I had glossed over this and didn't realize Typescript has intersection types.

Union would be safer since the union of _types_ only permits us to use the intersection of the types' _methods_. (Why?)
|Type|`eat()`|`shootWebs()`|`writeArticles()`|
|-|-|-|-|
|`Spider`|:white_check_mark:|:white_check_mark:||
|`Man`|:white_check_mark:||:white_check_mark:|
|`Spider \| Man`|:white_check_mark:|||
|`Spider & Man`|:white_check_mark:|:white_check_mark:|:white_check_mark:|

However we've been using TestFoo to _extend_ Foo's functionality with new functions (setters, etc.), so intersection types make sense for what we're currently doing.